app/controllers/users/sessions_controller.rb

```diff
@@ -3,13 +3,17 @@ class Users::SessionsController < Devise::SessionsController
   private
 
     def after_sign_in_path_for(resource)
-      if !verifying_via_email? && resource.show_welcome_screen?
+      if !verifying_via_email? && (resource.show_welcome_screen? || (skip_verification? && resource.first_login?))
         welcome_path
       else
         super
       end
     end
 
+    def skip_verification?
+      Setting["feature.user.skip_verification"]
```

Comment:
THere's a trailing `.present?` missing here, although this method won't be no longer needed here (and its a dup of the one at the verification concern